openhands/storage/data_models/user_secrets.py

```diff
@@ -136,3 +137,31 @@ def convert_dict_to_mappingproxy(
                 new_data['custom_secrets'] = secrets
 
         return new_data
+    
+
+    def set_event_stream_secrets(self, event_stream: EventStream) -> None:
```

Comment:
It feels a bit weird to me that a UserSecrets object knows about EventStreams. I think the reverse might be a better move? i.e. have `event_stream.set_secrets()` take in a SecretStore

Reply:
We I can do that; currently we also have `ProviderHandler` take in an event stream and mask the secrets

The idea is that `ProviderHandler` and `UserSecrets` know best how to manipulate their own values, so we let them handle it instead

Reply:
I have a nice solution to reverse the dependency but I'd like to do it in a separate PR (so we can revert it in the event refresh stops working or secrets are accidentally exposed)